sympy/core/assumptions.py

```diff
@@ -173,7 +173,7 @@
     'zero           ->  even & finite',
 
     'prime          ->  integer & positive',
-    'composite      ==  integer & positive & !prime',
+    'composite      ->  integer & positive & !prime',
```

Comment:
Then how about a hack for those assumptions so that when they are encountered during creation the assumptions `composite=True` is also added? Or an `is_composite` handler that takes over where the assumptions fail because of the `->` instead of the `==`?